### PR TITLE
Refactor/upload

### PR DESCRIPTION
**Pull request by phuens**
1. Updated the Ajax calls 
2. Removed unnecessary functions 
3. Added Documentation

### CODE REVIEW

File: lexos/static/js/scripts_upload.js [hunk 1]
```diff
@@ -1,197 +1,197 @@
 $(function () {
-  if ($('.fa-folder-open-o')[0].style.display != 'none' &&
-    $('.fa-folder-open-o')[0].innerText.indexOf('Response') != -1) {
-    alert('Steven\'s special bug detected! Click OK and the page will load.')
-    $('.fa-folder-open-o')[0].style.display = 'none'
-  }
-
+  /*message for when you hover on the open folder icon on the top right*/
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: move all the function definition outside of `$(function(){   })`

File: lexos/static/js/scripts_upload.js [hunk 2]
```diff
@@ -1,197 +1,197 @@
...
-      var added = 0
-
+    let files = e.target.files || e.dataTransfer.files
+    // total number of files uploaded
+    let totalFiles = files.length
+    //resets the progress bar to defualt message after a file upload
+    resetProgressBar()
+
+    const progress = $('#progress'),
+          progressBar = $('#progress-bar')
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: I don't think this is standard syntax, just use `const progressBar`

File: lexos/static/js/scripts_upload.js [hunk 3]
```diff
@@ -1,197 +1,197 @@
...
-  // file drag hover
-  function FileDragHover(e) {
-    e.stopPropagation()
-    e.preventDefault()
-    e.target.className = (e.type == 'dragover' ? 'hover' : '')
-  }
-
-  var numberOfFileDone = parseInt($('.fa-folder-open-o')[0].id)
-
+  let numberOfFileDone = parseInt($('.fa-folder-open-o')[0].id)
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: If you use let, please inform me in code comments the following:
1. why do you have to mutate the variable
2. where in the code did you mutated the variable.

File: lexos/static/js/scripts_upload.js [hunk 4]
```diff
@@ -1,197 +1,197 @@
...
   $('#uploadbrowse').click(function () {
     $('#fileselect').click()
   })
 
-  // ------------------- FILEDRAG -----------------------------
-
-  var allowedFileTypes = ['txt', 'xml', 'html', 'sgml', 'lexos']
-
+  //alias function
   function $id(id) {
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: I don't think you need this. `$(document.getElementById("id"))` is equivalent to `$('#id')`

File: lexos/static/js/scripts_upload.js [hunk 5]
```diff
@@ -1,197 +1,197 @@
...
+    // total number of files uploaded
+    let totalFiles = files.length
+    //resets the progress bar to defualt message after a file upload
+    resetProgressBar()
+
+    const progress = $('#progress'),
+          progressBar = $('#progress-bar')
+
+    //Process all File objects
+    for (let i = 0, f; f = files[i]; i++) {
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: `for (const f of files)`, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

File: lexos/static/js/scripts_upload.js [hunk 6]
```diff
@@ -1,197 +1,197 @@
...
+  if (window.File && window.FileList && window.FileReader) {
+    Init()
+  }
+})
+
 
-    if (AllowedFileType(file.name) && file.size <= $id('MAX_FILE_SIZE').value) {
-      if (file.size == 0) {
+/**
+ * @return {boolean}
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: This is not a documentation. You need to write the summary for this function and explain the parameter

File: lexos/static/js/scripts_upload.js [hunk 7]
```diff
@@ -1,197 +1,197 @@
...
+              }
+              else {
                 template.find('.file-information').find('.file-type').html(file.type)
-                var contents = e.target.result.replace(/</g, '&lt;')
-                  .replace(/>/g, '&gt;')
+                e.target.result.replace(/</g, '&lt;').replace(/>/g, '&gt;')
               }
-              var file_size = file.size
-              if (file.size < 1024) {
+              let file_size = file.size
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: `const fileSizeInByte`

File: lexos/static/js/scripts_upload.js [hunk 8]
```diff
@@ -1,197 +1,197 @@
...
+              if (file_size < 1024) {
                 template.find('.file-information').find('.file-size').html(file.size + 'bytes')
-              } else if (file.size < 1048576) {
+              }
+              else if (file.size < 1048576) {
                 file_size = (file.size / 1024).toFixed(1)
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: `const fileSizeInKB`

File: lexos/static/js/scripts_upload.js [hunk 9]
```diff
@@ -1,197 +1,197 @@
...
                 template.find('.file-information').find('.file-size').html(file_size + 'KB')
-              } else {
+              }
+              else {
                 file_size = (file.size / 1024 / 1024).toFixed(1)
```
Comment (chantisnake) [inline lexos/static/js/scripts_upload.js]: `const fileSizeInMB`

**Comment by chantisnake** [comment]
Also, you need to destruct `FileSelectHandler` into smaller functions.

File: lexos/static/js/scripts_upload.js [hunk 10]
```diff
@@ -1,197 +1,197 @@
...
+    if (xhr.upload) {
+      // file drop
+      filedrag.addEventListener('dragover', fileDragHover, false)
+      filedrag.addEventListener('dragleave', fileDragHover, false)
+      filedrag.addEventListener('drop', FileSelectHandler, false)
+    }
+  }
+
+  //Call initialization function
+  if (window.File && window.FileList && window.FileReader) {
```
Comment (Weiqi97) [inline lexos/static/js/scripts_upload.js]: Should this part be moved to ```$(function () {})```?

File: lexos/static/js/scripts_upload.js [hunk 11]
```diff
@@ -1,197 +1,197 @@
...
-    $('#progress').html('Ready For Files To Upload').css('color', '#074178').delay(3000).show()
+    progress.html('Ready For Files To Upload').css('color', '#074178').delay(3000).show()
     $id('fileselect').value = '' // this allows the event to fire on "change" in chrome. the value property changing is the
     // normal trigger, for some reason firefox overwrote this with their own behavior.
   }
 
-  // upload and display file contents
-  function UploadAndParseFile(file) {
-    var filename = file.name.replace(/ /g, '_')
+  // initialize
```
Comment (Weiqi97) [inline lexos/static/js/scripts_upload.js]: The documentation should be in format of
```javascript
/**
 * Initialize (Try to be more specific here..)
 * @returns {void}
```

File: lexos/static/js/scripts_upload.js [hunk 12]
```diff
@@ -1,197 +1,197 @@
...
+}
+
+
+// upload and display file contents
+function UploadAndParseFile(file,fileSize) {
+    let filename = file.name.replace(/ /g, '_')
+
+    if (AllowedFileType(file.name) && file.size <= fileSize) {
+      if (file.size === 0) {
         alert('Cannot process blank file -- ' + file.name)
```
Comment (Weiqi97) [inline lexos/static/js/scripts_upload.js]: Maybe try template literal here..
```javascript 
`Cannot process blank file -- ${file.name}`
```

**Comment by Weiqi97** [comment]
There are a couple more places you could use [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) to improve readability. Also this page uses alert instead of bootstrap modal. Maybe change them so the page is consistent with others.

**Comment by chantisnake** [comment]
Also, you need to add `upload.js` to eslint.

**Pull request merged**